var/spack/repos/builtin/packages/slepc/package.py

```diff
@@ -156,19 +158,20 @@ def run_hello_test(self):
         """Run stand alone test: hello"""
         test_dir = self.test_suite.current_test_data_dir
 
-        if not os.path.exists(test_dir):
-            print('Skipping slepc test')
+        if not os.path.isfile(join_path(test_dir, 'hello.c')):
+            tty.msg('Skipping slepc test: failed to find hello.c')
             return
 
         exe = 'hello'
         cc_exe = os.environ['CC']
 
         self.run_test(exe=cc_exe,
                       options=['-I{0}'.format(self.prefix.include),
-                               '-L', self.prefix.lib, '-l', 'slepc',
-                               '-L', self.spec['petsc'].prefix.lib, '-l', 'petsc',
-                               '-L', self.spec['mpi'].prefix.lib, '-l', 'mpi',
-                               '-o', exe, join_path(test_dir, 'hello.c')],
+                               '-L{0}'.format(self.prefix.lib),
+                               '-L{0}'.format(self.spec['petsc'].prefix.lib),
+                               '-L{0}'.format(self.spec['mpi'].prefix.lib),
```

Comment:
I'm not familiar with this library, but I'd make sure that `spec['mpi']` is always going to be defined (and not some other lib providing mpi).

Reply:
The `mpi` package is virtual so automatically "references" the package that provides the interface.

Reply:
perfecto!

Reply:
We have an explicit example of this to find the name of the providing package in the docs at https://spack-tutorial.readthedocs.io/en/latest/tutorial_packaging.html#querying-spec-names.

Reply:
I understand this is perhaps work in progress - but would be a good idea to replicate the code from PETSc as in https://github.com/spack/spack/pull/27597

i.e:
-  stage in src/eps/tests
- compile test code with the provided makefile
- run the test code...
etc...

You can check slepc/makefile check_build: target - for some tests that can be run and src/eps/tests for the run commands

Using the example sources and makefile will avoid the need for manually constructing the compile command [as above] that likely won't always work